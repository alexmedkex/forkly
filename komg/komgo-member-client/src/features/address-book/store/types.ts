import { List } from 'immutable'
import { ImmutableMap } from '../../../utils/types'
import { Action } from 'redux'
import { IMember } from '../../members/store/types'

export enum AddressBookActionType {
  GET_COMPANIES_REQUEST = '@@addressBook/GET_COMPANIESS_REQUEST',
  GET_COMPANIES_SUCCESS = '@@addressBook/GET_COMPANIESS_SUCCESS',
  GET_COMPANIES_FAILURE = '@@addressBook/GET_COMPANIESS_FAILURE',
  GET_COMPANY_REQUEST = '@@addressBook/GET_COMPANY_REQUEST',
  GET_COMPANY_SUCCESS = '@@addressBook/GET_COMPANY_SUCCESS',
  GET_COMPANY_FAILURE = '@@addressBook/GET_COMPANY_FAILURE',
  GENERATE_MEMBER_PACKAGE_REQUEST = '@@addressBook/GENERATE_MEMBER_PACKAGE_REQUEST',
  GENERATE_MEMBER_PACKAGE_SUCCESS = '@@addressBook/GENERATE_MEMBER_PACKAGE_SUCCESS',
  GENERATE_MEMBER_PACKAGE_FAILURE = '@@addressBook/GENERATE_MEMBER_PACKAGE_FAILURE',
  ADD_COMPANY_TO_ENS_REQUEST = '@@addressBook/ADD_COMPANY_TO_ENS_REQUEST',
  ADD_COMPANY_TO_ENS_SUCCESS = '@@addressBook/ADD_COMPANY_TO_ENS_SUCCESS',
  ADD_COMPANY_TO_ENS_FAILURE = '@@addressBook/ADD_COMPANY_TO_ENS_FAILURE',
  CONFIGURE_MQ_REQUEST = '@@addressBook/CONFIGURE_MQ_REQUEST',
  CONFIGURE_MQ_SUCCESS = '@@addressBook/CONFIGURE_MQ_SUCCESS',
  CONFIGURE_MQ_FAILURE = '@@addressBook/CONFIGURE_MQ_FAILURE',
  CREATE_COMPANY_REQUEST = '@@addressBook/CREATE_COMPANY_REQUEST',
  CREATE_COMPANY_SUCCESS = '@@addressBook/CREATE_COMPANY_SUCCESS',
  CREATE_COMPANY_FAILURE = '@@addressBook/CREATE_COMPANY_FAILURE',
  UPDATE_COMPANY_REQUEST = '@@addressBook/UPDATE_COMPANY_REQUEST',
  UPDATE_COMPANY_SUCCESS = '@@addressBook/UPDATE_COMPANY_SUCCESS',
  UPDATE_COMPANY_FAILURE = '@@addressBook/UPDATE_COMPANY_FAILURE',
  DeleteCompanyByIdSuccess = '@@addressBook/DELETE_COMPANY_BY_ID_SUCCESS'
}

export interface AddressBookStateProperties {
  companies: List<IMember>
}

export type AddressBookState = ImmutableMap<AddressBookStateProperties>

export interface GetCompaniesRequest extends Action {
  type: AddressBookActionType.GET_COMPANIES_SUCCESS
  payload: IMember[]
}
export interface GetCompanyRequest extends Action {
  type: AddressBookActionType.GET_COMPANY_SUCCESS
  payload: IMember
}
export interface GenerateMemberPackageRequest extends Action {
  type: AddressBookActionType.GENERATE_MEMBER_PACKAGE_SUCCESS
  payload: IMember
}
export interface AddCompanyToENSRequest extends Action {
  type: AddressBookActionType.ADD_COMPANY_TO_ENS_SUCCESS
  payload: IMember
}
export interface ConfigureMQRequest extends Action {
  type: AddressBookActionType.CONFIGURE_MQ_SUCCESS
  payload: IMember
}

export interface CreateCompanyRequest extends Action {
  type: AddressBookActionType.CREATE_COMPANY_SUCCESS
  payload: IMember
}
export interface UpdateCompanyRequest extends Action {
  type: AddressBookActionType.UPDATE_COMPANY_SUCCESS
  payload: IMember
}

export interface DeleteCompanyByIdSuccess extends Action {
  type: AddressBookActionType.DeleteCompanyByIdSuccess
  payload: { id: string }
}

export type AddressBookActions =
  | GetCompaniesRequest
  | GetCompanyRequest
  | CreateCompanyRequest
  | UpdateCompanyRequest
  | GenerateMemberPackageRequest
  | AddCompanyToENSRequest
  | ConfigureMQRequest
  | DeleteCompanyByIdSuccess
