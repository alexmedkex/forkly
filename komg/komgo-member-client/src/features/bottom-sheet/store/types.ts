import { Action } from 'redux'
import { ImmutableMap } from '../../../utils/types'
import {
  CreateDocumentError,
  CreateDocumentSuccess,
  DocumentRegisteredError,
  DocumentRegisteredSuccess,
  HasId,
  HasName
} from '../../document-management/store/types'
import { ActionCreator } from 'react-redux'

export enum BottomSheetActionType {
  REMOVE_BOTTOMSHEET_ITEM = '@@btsh/REMOVE_BOTTOMSHEET_ITEM',
  RETRY_BOTTOMSHEET_ITEM = '@@btsh/RETRY_BOTTOMSHEET_ITEM',
  UPDATE_BOTTOMSHEET_ITEM = '@@btsh/UPDATE_BOTTOMSHEET_ITEM',
  CREATE_BOTTOMSHEET_ITEM = '@@btsh/CREATE_BOTTOMSHEET_ITEM'
}

export interface BottomSheetItem extends HasId, HasName {
  state: BottomSheetStatus
  itemType?: BottomSheetItemType
  displayStatus?: string
  action?: ActionCreator<any>
  navigationLink?: string
}

export enum BottomSheetItemType {
  REGISTER_KYC_DOCUMENT,
  CREATE_MAGIC_LINK,
  REGISTER_RD_DOCUMENT,
  REGISTER_TRADE_DOCUMENT
}

export enum BottomSheetStatus {
  NEW = 'NEW',
  PENDING = 'PENDING',
  REGISTERED = 'REGISTERED',
  FAILED = 'FAILED'
}

export interface BottomSheetStateFields {
  visible: boolean
  items: BottomSheetItem[]
}

export interface RemoveBottomSheetItem extends Action {
  type: BottomSheetActionType.REMOVE_BOTTOMSHEET_ITEM
  payload: HasId
}

export interface RetryRegisterItem extends Action {
  type: BottomSheetActionType.RETRY_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface UpdateBottomSheetItem extends Action {
  type: BottomSheetActionType.UPDATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface CreateBottomSheetItem extends Action {
  type: BottomSheetActionType.CREATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export type BottomSheetState = ImmutableMap<BottomSheetStateFields>

export type BottomSheetAction =
  | DocumentRegisteredSuccess
  | DocumentRegisteredError
  | CreateDocumentSuccess
  | CreateDocumentError
  | RemoveBottomSheetItem
  | RetryRegisterItem
  | UpdateBottomSheetItem
  | CreateBottomSheetItem
