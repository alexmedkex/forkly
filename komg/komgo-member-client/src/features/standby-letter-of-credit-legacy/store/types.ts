import { ImmutableMap, stringOrNull } from '../../../utils/types'
import { ActionType } from '../../letter-of-credit-legacy/store/types'
import { List, Map } from 'immutable'
import { IStandbyLetterOfCredit, TradeSource } from '@komgo/types/dist'

export enum StandbyLetterOfCreditActionType {
  SUBMIT_STANDBY_LETTER_OF_CREDIT_REQUEST = '@@standby-letters-of-credit/SUBMIT_LETTER_OF_CREDIT_REQUEST',
  SUBMIT_STANDBY_LETTER_OF_CREDIT_SUCCESS = '@@standby-letters-of-credit/SUBMIT_LETTER_OF_CREDIT_SUCCESS',
  SUBMIT_STANDBY_LETTER_OF_CREDIT_FAILURE = '@@standby-letters-of-credit/SUBMIT_LETTER_OF_CREDIT_FAILURE',
  GET_STANDBY_LETTER_OF_CREDIT_REQUEST = '@@standby-letters-of-credit/GET_STANDBY_LETTER_OF_CREDIT_REQUEST',
  GET_STANDBY_LETTER_OF_CREDIT_SUCCESS = '@@standby-letters-of-credit/GET_STANDBY_LETTER_OF_CREDIT_SUCCESS',
  GET_STANDBY_LETTER_OF_CREDIT_FAILURE = '@@standby-letters-of-credit/GET_STANDBY_LETTER_OF_CREDIT_FAILURE',
  FETCH_STANDBY_LETTERS_OF_CREDIT_REQUEST = '@@standby-letters-of-credit/FETCH_STANDBY_LETTERS_OF_CREDIT_REQUEST',
  FETCH_STANDBY_LETTERS_OF_CREDIT_SUCCESS = '@@standby-letters-of-credit/FETCH_STANDBY_LETTERS_OF_CREDIT_SUCCESS',
  FETCH_STANDBY_LETTERS_OF_CREDIT_FAILURE = '@@standby-letters-of-credit/FETCH_STANDBY_LETTERS_OF_CREDIT_FAILURE',
  FETCH_STANDBY_LETTER_OF_CREDIT_DOCUMENTS_REQUEST = '@@standby-letters-of-credit/FETCH_STANDBY_LETTER_OF_CREDIT_DOCUMENTS_REQUEST',
  FETCH_STANDBY_LETTER_OF_CREDIT_DOCUMENTS_SUCCESS = '@@standby-letters-of-credit/FETCH_STANDBY_LETTER_OF_CREDIT_DOCUMENTS_SUCCESS',
  FETCH_STANDBY_LETTER_OF_CREDIT_DOCUMENTS_FAILURE = '@@standby-letters-of-credit/FETCH_STANDBY_LETTER_OF_CREDIT_DOCUMENTS_FAILURE',
  ISSUE_STANDBY_LETTER_OF_CREDIT_REQUEST = '@@standby-letters-of-credit/ISSUE_STANDBY_LETTER_OF_CREDIT_REQUEST',
  ISSUE_STANDBY_LETTER_OF_CREDIT_SUCCESS = '@@standby-letters-of-credit/ISSUE_STANDBY_LETTER_OF_CREDIT_SUCCESS',
  ISSUE_STANDBY_LETTER_OF_CREDIT_FAILURE = '@@standby-letters-of-credit/ISSUE_STANDBY_LETTER_OF_CREDIT_FAILURE',
  REJECT_STANDBY_LETTER_OF_CREDIT_REQUEST = '@@standby-letters-of-credit/REJECT_STANDBY_LETTER_OF_CREDIT_REQUEST',
  REJECT_STANDBY_LETTER_OF_CREDIT_SUCCESS = '@@standby-letters-of-credit/REJECT_STANDBY_LETTER_OF_CREDIT_SUCCESS',
  REJECT_STANDBY_LETTER_OF_CREDIT_FAILURE = '@@standby-letters-of-credit/REJECT_STANDBY_LETTER_OF_CREDIT_FAILURE'
}

export interface ITradeId {
  source: TradeSource
  sourceId: string
}

export interface StandbyLetterOfCreditStateProperties {
  byId: Map<string, IStandbyLetterOfCredit>
  ids: List<string>
  total: number
}

export interface FetchStandByLettersOfCredit {
  filter: {
    options: any | undefined
  }
  polling?: boolean
}

export type StandbyLetterOfCreditState = ImmutableMap<StandbyLetterOfCreditStateProperties>
