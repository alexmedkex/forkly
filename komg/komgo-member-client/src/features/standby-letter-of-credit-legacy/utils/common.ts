import { IStandbyLetterOfCredit, StandbyLetterOfCreditStatus } from '@komgo/types'
export const isRejectedStandbyLetterOfCredit = (letter: IStandbyLetterOfCredit) => {
  return (
    !!letter &&
    [StandbyLetterOfCreditStatus.RequestRejected, StandbyLetterOfCreditStatus.IssuedRejected].includes(letter.status)
  )
}
