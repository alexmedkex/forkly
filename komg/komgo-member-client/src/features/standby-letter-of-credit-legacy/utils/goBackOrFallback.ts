import { history } from '../../../store'

export const goBackOrFallBackTo = ({
  defaultHistoryLength = 4, // In KOMGO initial history.length is by default 4 because the login redirects and route configuration
  fallbackURL
}: {
  defaultHistoryLength?: number
  fallbackURL: string
}) => {
  // WORKAROUND check if the history length is the default defaultHistoryLength
  // TODO LS remove anything that mimic browser behaviors (e.g. back). This approach is fragile
  if (history.length <= defaultHistoryLength) {
    history.push(fallbackURL)
  } else {
    history.goBack()
  }
}
