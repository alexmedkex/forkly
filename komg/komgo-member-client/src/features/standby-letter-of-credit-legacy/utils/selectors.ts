import { Task, TaskStatus } from '../../tasks/store/types'

export const findTasksByStandByLetterOfCreditAndStatuses = (
  letterId: string,
  tasks: Task[],
  statuses: TaskStatus[]
) => {
  return tasks.filter(task => task.context && task.context.sblcStaticId === letterId && statuses.includes(task.status))
}
