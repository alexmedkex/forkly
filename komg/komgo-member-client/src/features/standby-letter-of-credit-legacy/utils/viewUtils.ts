import { ReviewDecision, IssueFormValues } from '../components/issue-form'
import { StandbyLetterOfCreditTaskType, IStandbyLetterOfCredit } from '@komgo/types'
import { IMember } from '../../members/store/types'

export const findActiveFields = (
  taskType: StandbyLetterOfCreditTaskType,
  reviewDecision: ReviewDecision
): Array<Partial<keyof IStandbyLetterOfCredit>> =>
  taskType === StandbyLetterOfCreditTaskType.ReviewRequested && reviewDecision === ReviewDecision.ApproveApplication
    ? ['issuingBankReference', 'issuingBankPostalAddress']
    : []

const isSet = (val?: string) => !!val

// cyclomatic complexity....
const isApproveApplicationUseCase = (
  taskIsReviewRequested: boolean,
  reviewDecision: ReviewDecision,
  hasIssuingBankReference: boolean,
  hasIssuingBankPostalAddress: boolean,
  beneficiary: IMember,
  document?: File
) => {
  const rules = [
    taskIsReviewRequested &&
      reviewDecision === ReviewDecision.ApproveApplication &&
      hasIssuingBankReference &&
      hasIssuingBankPostalAddress &&
      beneficiary.isMember,
    taskIsReviewRequested &&
      reviewDecision === ReviewDecision.ApproveApplication &&
      !beneficiary.isMember &&
      document &&
      hasIssuingBankReference &&
      hasIssuingBankPostalAddress
  ]

  return rules.includes(true)
}

export const isSubmitResponseEnabled = (
  values: IssueFormValues,
  beneficiary: IMember,
  taskType: StandbyLetterOfCreditTaskType
) => {
  const { document, reviewDecision, standbyLetterOfCredit } = values

  const taskIsReviewRequested = taskType === StandbyLetterOfCreditTaskType.ReviewRequested

  const hasIssuingBankPostalAddress = isSet(standbyLetterOfCredit.issuingBankPostalAddress)
  const hasIssuingBankReference = isSet(standbyLetterOfCredit.issuingBankReference)

  const rules = [
    taskIsReviewRequested && reviewDecision === ReviewDecision.RejectApplication,
    isApproveApplicationUseCase(
      taskIsReviewRequested,
      reviewDecision,
      hasIssuingBankReference,
      hasIssuingBankPostalAddress,
      beneficiary,
      document
    )
  ]
  return rules.includes(true)
}

export const viewSubmitHandler = (
  values: IssueFormValues,
  taskType: StandbyLetterOfCreditTaskType,
  issueStandbyLetterOfCredit: (standbyLetterOfCredit: IStandbyLetterOfCredit, file: File) => void,
  rejectStandbyLetterOfCreditRequest: (staticId: string, issuingBankReference: string) => void
) => {
  const taskIsReviewRequested = taskType === StandbyLetterOfCreditTaskType.ReviewRequested

  if (taskIsReviewRequested && values.reviewDecision === ReviewDecision.ApproveApplication) {
    issueStandbyLetterOfCredit(values.standbyLetterOfCredit, values.document)
  } else if (taskIsReviewRequested && values.reviewDecision === ReviewDecision.RejectApplication) {
    rejectStandbyLetterOfCreditRequest(values.standbyLetterOfCredit.staticId, values.rejectionReference)
  }
}
