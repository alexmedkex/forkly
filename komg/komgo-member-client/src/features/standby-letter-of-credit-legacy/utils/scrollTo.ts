import scrollIntoView from 'scroll-into-view'
import { history } from '../../../store/history'
import { SIZES } from '../../../utils/media'

export const scrollTo = (selector: string) => {
  const isMobile = window.innerWidth < SIZES.desktop

  if (isMobile) {
    return
  }

  const target = document.querySelector(selector)
  scrollIntoView(target, { time: 500, align: { top: 0.1 } })
}
