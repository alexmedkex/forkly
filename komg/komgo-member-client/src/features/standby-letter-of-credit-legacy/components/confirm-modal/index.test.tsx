import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { ConfirmModalProps, Confirm } from './index'
import { fakeMember, fakeTrade } from '../../../letter-of-credit-legacy/utils/faker'
import { ServerError } from '../../../../store/common/types'
import { mount } from 'enzyme'

// the magic https://github.com/Semantic-Org/Semantic-UI-React/issues/2454#issuecomment-373246622
jest.mock('semantic-ui-react/dist/commonjs/addons/Portal/Portal', () => ({ children }) => children)

describe('Confirm', () => {
  let props: ConfirmModalProps

  beforeEach(() => {
    const trade = fakeTrade()
    const issuingBank = fakeMember({ isFinancialInstitution: true })
    props = {
      errors: [],
      isSubmitting: false,
      open: true,
      onCancel: jest.fn(),
      onSubmit: jest.fn(),
      title: 'Submit SBLC Application',
      children: (
        <p>
          You are about to submit an SBLC application for the financing of trade {trade.buyerEtrmId} to{' '}
          {issuingBank.x500Name.CN}
        </p>
      )
    }
  })

  describe('render', () => {
    it('shows the UI', () => {
      expect(renderer.create(<Confirm {...props} />).toJSON()).toMatchSnapshot()
    })

    it('shows is submitting', () => {
      expect(renderer.create(<Confirm {...props} isSubmitting={true} />).toJSON()).toMatchSnapshot()
    })

    it('shows an error', () => {
      const error: ServerError = {
        message: 'Error',
        errorCode: 'E001',
        requestId: '5d9a7aaf-f55a-431c-9d26-8204d9ecdbe9',
        origin: 'trade-finance',
        fields: {}
      }
      expect(renderer.create(<Confirm {...props} errors={[error]} />).toJSON()).toMatchSnapshot()
    })
    describe('submit button', () => {
      it('has the form set to formId if formId provided', () => {
        expect(renderer.create(<Confirm {...props} formId={'myForm'} />).toJSON()).toMatchSnapshot()
      })
    })
  })

  describe('onSubmit', () => {
    it('calls the hanlder if defined', () => {
      const wrapper = mount(<Confirm {...props} />)
      const button = wrapper.find('button[data-test-id="submit-confirm-button"]').first()
      button.simulate('click')
      expect(props.onSubmit).toHaveBeenCalled()
    })
  })

  describe('onCancel', () => {
    it('calls the hanlder', () => {
      const wrapper = mount(<Confirm {...props} />)
      const button = wrapper.find('button[data-test-id="submit-cancel-button"]').first()
      button.simulate('click')
      expect(props.onCancel).toHaveBeenCalled()
    })
  })
})
