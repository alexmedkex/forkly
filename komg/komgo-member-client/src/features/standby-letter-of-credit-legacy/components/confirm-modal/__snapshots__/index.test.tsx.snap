// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm render shows an error 1`] = `
<div
  className="ui page modals dimmer transition visible active"
>
  <div
    className="ui small modal transition visible active"
    style={
      Object {
        "marginTop": undefined,
      }
    }
  >
    <div
      className="header"
    >
      Submit SBLC Application
    </div>
    <div
      className="content"
    >
      <div
        className="ui negative message"
        data-test-id="errorMessage"
      >
        <div
          className="header"
        >
          Submit SBLC Application
        </div>
        <p>
          Error
        </p>
      </div>
    </div>
    <div
      className="actions"
    >
      <button
        className="ui button"
        data-test-id="submit-cancel-button"
        onClick={[Function]}
        role="button"
      >
        Cancel
      </button>
      <button
        className="ui primary button"
        data-test-id="submit-confirm-button"
        onClick={[Function]}
        role="button"
        type="submit"
      >
        Confirm
      </button>
    </div>
  </div>
</div>
`;

exports[`Confirm render shows is submitting 1`] = `
.c1 {
  display: inline-block;
  font-size: 14px;
  font-weight: normal;
  color: #142435;
  margin: 0;
}

.c0 {
  margin-top: 0;
  position: relative;
  background-color: rgba(255,255,255,0.85);
}

<div
  className="ui page modals dimmer transition visible active"
>
  <div
    className="ui small modal transition visible active"
    style={
      Object {
        "marginTop": undefined,
      }
    }
  >
    <div
      className="header"
    >
      Submit SBLC Application
    </div>
    <div
      className="content"
    >
      <div
        className="c0"
        data-test-id="loadingTransition"
      >
        <img
          className="ui mini centered image"
          src="/images/bars.svg"
        />
        <div
          className="ui header centered"
        >
          <span
            className="c1"
            color="#142435"
          >
            Submitting...
          </span>
        </div>
      </div>
    </div>
    <div
      className="actions"
    >
      <button
        className="ui disabled button"
        data-test-id="submit-cancel-button"
        disabled={true}
        onClick={[Function]}
        role="button"
        tabIndex={-1}
      >
        Cancel
      </button>
      <button
        className="ui primary disabled button"
        data-test-id="submit-confirm-button"
        disabled={true}
        onClick={[Function]}
        role="button"
        tabIndex={-1}
        type="submit"
      >
        Confirm
      </button>
    </div>
  </div>
</div>
`;

exports[`Confirm render shows the UI 1`] = `
<div
  className="ui page modals dimmer transition visible active"
>
  <div
    className="ui small modal transition visible active"
    style={
      Object {
        "marginTop": undefined,
      }
    }
  >
    <div
      className="header"
    >
      Submit SBLC Application
    </div>
    <div
      className="content"
    >
      <p>
        You are about to submit an SBLC application for the financing of trade 
        E222
         to
         
        Applicant Name
      </p>
    </div>
    <div
      className="actions"
    >
      <button
        className="ui button"
        data-test-id="submit-cancel-button"
        onClick={[Function]}
        role="button"
      >
        Cancel
      </button>
      <button
        className="ui primary button"
        data-test-id="submit-confirm-button"
        onClick={[Function]}
        role="button"
        type="submit"
      >
        Confirm
      </button>
    </div>
  </div>
</div>
`;

exports[`Confirm render submit button has the form set to formId if formId provided 1`] = `
<div
  className="ui page modals dimmer transition visible active"
>
  <div
    className="ui small modal transition visible active"
    style={
      Object {
        "marginTop": undefined,
      }
    }
  >
    <div
      className="header"
    >
      Submit SBLC Application
    </div>
    <div
      className="content"
    >
      <p>
        You are about to submit an SBLC application for the financing of trade 
        E222
         to
         
        Applicant Name
      </p>
    </div>
    <div
      className="actions"
    >
      <button
        className="ui button"
        data-test-id="submit-cancel-button"
        onClick={[Function]}
        role="button"
      >
        Cancel
      </button>
      <button
        className="ui primary button"
        data-test-id="submit-confirm-button"
        form="myForm"
        onClick={[Function]}
        role="button"
        type="submit"
      >
        Confirm
      </button>
    </div>
  </div>
</div>
`;
